dev/tools/update_icons.dart

```diff
@@ -152,71 +155,135 @@ void main(List<String> args) {
   if (path.basename(Directory.current.path) == 'tools')
     Directory.current = Directory.current.parent.parent;
 
-  final ArgParser argParser = ArgParser();
-  argParser.addOption(kOptionCodepointsPath, defaultsTo: kDefaultCodepointsPath);
-  argParser.addOption(kOptionIconsPath, defaultsTo: kDefaultIconsPath);
-  argParser.addFlag(kOptionDryRun, defaultsTo: false);
-  final ArgResults argResults = argParser.parse(args);
+  final ArgResults argResults = _handleArguments(args);
 
-  final File iconFile = File(path.absolute(argResults[kOptionIconsPath] as String));
-  if (!iconFile.existsSync()) {
-    stderr.writeln('Icons file not found: ${iconFile.path}');
+  final File iconClassFile = File(path.normalize(path.absolute(argResults[_iconsClassPathOption] as String)));
+  if (!iconClassFile.existsSync()) {
+    stderr.writeln('Error: Icons file not found: ${iconClassFile.path}');
+    exit(1);
+  }
+  final File newCodepointsFile = File(path.absolute(path.normalize(argResults[_newCodepointsPathOption] as String)));
+  if (!newCodepointsFile.existsSync()) {
+    stderr.writeln('Error: New codepoints file not found: ${newCodepointsFile.path}');
     exit(1);
   }
-  final File codepointsFile = File(path.absolute(argResults[kOptionCodepointsPath] as String));
-  if (!codepointsFile.existsSync()) {
-    stderr.writeln('Codepoints file not found: ${codepointsFile.path}');
+  final File oldCodepointsFile = File(path.absolute(argResults[_oldCodepointsPathOption] as String));
+  if (!oldCodepointsFile.existsSync()) {
+    stderr.writeln('Error: Old codepoints file not found: ${oldCodepointsFile.path}');
     exit(1);
   }
 
-  final String iconData = iconFile.readAsStringSync();
-  final String codepointData = codepointsFile.readAsStringSync();
-  final String newIconData = regenerateIconsFile(iconData, codepointData);
+  final String newCodepointsString = newCodepointsFile.readAsStringSync();
+  final Map<String, String> newTokenPairMap = _stringToTokenPairMap(newCodepointsString);
+
+  final String oldCodepointsString = oldCodepointsFile.readAsStringSync();
+  final Map<String, String> oldTokenPairMap = _stringToTokenPairMap(oldCodepointsString);
 
-  if (argResults[kOptionDryRun] as bool)
+  _testIsMapSuperset(newTokenPairMap, oldTokenPairMap);
+
+  final String iconClassFileData = iconClassFile.readAsStringSync();
+
+  stderr.writeln('Generating new token pairs.');
+  final String newIconData = _regenerateIconsFile(iconClassFileData, newTokenPairMap);
+
+  if (argResults[_dryRunOption] as bool) {
     stdout.writeln(newIconData);
-  else
-    iconFile.writeAsStringSync(newIconData);
+  } else {
+    stderr.writeln('\nWriting to ${iconClassFile.path}.');
+    iconClassFile.writeAsStringSync(newIconData);
```

Comment:
If this fails, I assume we're OK with the default exception handler's report?

Reply:
Yeah. I don't think it can be improved.